doc/design/vic-machine/update.md

```diff
@@ -0,0 +1,207 @@
+# VCH Update (P0)
+This document defines the VCH update command line, and the related changes in vic-machine, to make VCH update process more reasonable.
+
+Here is the key points considerred in this Definition
+- Batch Operation
+
+  We'd like to update VCH configurations more than one item at one time to make the command easy to use
+- Reuse all options used in vic-machine create
+
+  In vic-machine create, we already have 57 options to define VCH configurations, though not all of them can be updated, quite amount of them should be able to be changed, and this number will grow up over time, so we definitely do not want to introduce 3X that number of options to update (update means add/modify/remove)
+
+## Options
+First option is to introduce update primitives
+
+### Delta Options
+#### Update Primitives
+
+- add
```

Comment:
This approach is procedural, however the current vic-machine options are predominantly declarative.
This is going to result in a very large matrix of possible combinations which will be extremely awkward to vet and test.

I suggest using a different approach:
0. *DO NOT SUPPORT DELETE AT THIS TIME* - it's just fraught
1. extract the current VCH configuration into installData (or similar base struct - this is the in memory representation of the manifest concept).
2. run through the `reconfigure` options using the same parsing that we have for create (basically no new code) to generate a second installData
3. provide code to combine two installData's
  a. for elements with names (e.g. container-net, volume-store) if there's a value provided against that name in the second installData it's a modify op
  b. for elements without names (e.g. --cert) if there's a value provided it's a replacement
  c. _maybe_ have some `reconfigure` specific arguments for those options that generate lists: `--clear-registry-cas`, `--clear-tls-cas` to allow a means to remove existing entries (the default list argument behaviour being to append) that would be specified before new `--registry-ca` options
4. run the validator against the combined configuration
  a. this should include the validation for ip address range conflicts, etc, which are currently inlined in the argument processing.